### PR TITLE
Support for AutoDeleteOnIdle and DefaultMessageTimeToLive in Azure SB Queues

### PR DESCRIPTION
**Pull request by trbenning**
Resolves issue #39.

### CODE REVIEW

**Comment by ejsmith** [comment]
This looks good to me.

File: src/Azure/Queues/AzureServiceBusQueue.cs [hunk 1]
```diff
@@ -21,22 +21,33 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
...
+        public AzureServiceBusQueue(string connectionString, string queueName = null, int retries = 2, TimeSpan? workItemTimeout = null, bool shouldRecreate = false, RetryPolicy retryPolicy = null, ISerializer serializer = null, IEnumerable<IQueueBehavior<T>> behaviors = null, ILoggerFactory loggerFactory = null, TimeSpan? autoDeleteOnIdle = null, TimeSpan? defaultMessageTimeToLive = null) : base(serializer, behaviors, loggerFactory) {
             _queueName = queueName ?? typeof(T).Name;
             _namespaceManager = NamespaceManager.CreateFromConnectionString(connectionString);
             _retries = retries;
             if (workItemTimeout.HasValue && workItemTimeout.Value < TimeSpan.FromMinutes(5))
                 _workItemTimeout = workItemTimeout.Value;
 
+            if (autoDeleteOnIdle.HasValue && autoDeleteOnIdle.Value >= TimeSpan.FromMinutes(5))
+                _autoDeleteOnIdle = autoDeleteOnIdle.Value;
```
Comment (niemyjski) [inline src/Azure/Queues/AzureServiceBusQueue.cs]: Any reason why we'd want to limit this to timespans greater than 5 minutes? What happens if your app is idle shortly after startup?

Comment (jamie94bc) [inline src/Azure/Queues/AzureServiceBusQueue.cs]: That might be a requirement of Azure like `workItemTimeout.Value < TimeSpan.FromMinutes(5)` @trbenning?

Comment (trbenning) [inline src/Azure/Queues/AzureServiceBusQueue.cs]: @jamie94bc  is right. The minimum value for AutoDeleteOnIdle is 5 minutes.

File: src/Azure/Queues/AzureServiceBusQueue.cs [hunk 2]
```diff
@@ -54,6 +65,14 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     changes = true;
                 }
 
+                if (_queueDescription.AutoDeleteOnIdle != _autoDeleteOnIdle) {
+                    _queueDescription.AutoDeleteOnIdle = _autoDeleteOnIdle;
+                }
```
Comment (niemyjski) [inline src/Azure/Queues/AzureServiceBusQueue.cs]: @trbenning I think we are missing the changes = true check as in the lines above. Can you please confirm this.

Comment (trbenning) [inline src/Azure/Queues/AzureServiceBusQueue.cs]: Good catch. I've updated the source branch.

**Comment by niemyjski** [comment]
Thanks!

**Pull request merged**